pkg/extract/oracle/oracle.go

```diff
@@ -155,15 +161,27 @@ func Extract(inputPath string, opts ...Option) error {
 }
 
 func extract(def oracle.Definition, tos tos) (dataTypes.Data, error) {
+	id := func() string {
+		for _, r := range def.Metadata.Reference {
+			if r.Source == "elsa" {
+				return r.RefID
+			}
+		}
+		return ""
+	}()
+	if id == "" {
+		return dataTypes.Data{}, errors.Errorf("advisory ID not found. definition: %s", def.ID)
+	}
```

Comment:
To convert from a definition id, one can write
```suggestion
	var ss []string
	s := strings.TrimPrefix(def.ID, "oval:com.oracle.elsa:def:")
	for i := 0; i < len(s); i += 4 {
		end := i + 4
		if end > len(s) {
			end = len(s)
		}
		ss = append(ss, s[i:end])
	}
	if len(ss) < 2 {
		return dataTypes.Data{}, errors.Errorf("unexpected definition id. expected: %q, actual: %q", "oval:com.oracle.elsa:def:<yyyy>\\d{4,}", def.ID)
	}
	if _, err := time.Parse("2006", ss[0]); err != nil {
		return dataTypes.Data{}, errors.Errorf("unexpected definition id. expected: %q, actual: %q", "oval:com.oracle.elsa:def:<yyyy>\\d{4,}", def.ID)
	}
	id := fmt.Sprintf("ELSA-%s", strings.Join(ss, "-"))
```

Reply:
Which is preferable: searching by reference or converting from definition id?

Reply:
It seems that it is not always possible to divide it into four digits like `ELSA-2024-20865`, so this proposal is rejected.
https://linux.oracle.com/errata/ELSA-2024-20865.html

Reply:
```diff
:100644 100644 56c7f0f0 00000000 M	pkg/extract/oracle/oracle.go

diff --git a/pkg/extract/oracle/oracle.go b/pkg/extract/oracle/oracle.go
index 56c7f0f0..36cc46a2 100644
--- a/pkg/extract/oracle/oracle.go
+++ b/pkg/extract/oracle/oracle.go
@@ -161,16 +161,9 @@ func Extract(inputPath string, opts ...Option) error {
 }
 
 func extract(def oracle.Definition, tos tos) (dataTypes.Data, error) {
-	id := func() string {
-		for _, r := range def.Metadata.Reference {
-			if r.Source == "elsa" {
-				return r.RefID
-			}
-		}
-		return ""
-	}()
-	if id == "" {
-		return dataTypes.Data{}, errors.Errorf("advisory ID not found. definition: %s", def.ID)
+	id, _, ok := strings.Cut(strings.TrimSpace(def.Metadata.Title), ":")
+	if !ok {
+		return dataTypes.Data{}, errors.Errorf("unexpected title format. expected: %q, actual: %q", "<Advisory ID>: ...", def.Metadata.Title)
 	}
 
 	ds, err := collectPackages(def.Criteria, tos)

```

Reply:
There exists 5-degit ELSA ID...

```
{
  "id": "oval:com.oracle.elsa:def:202420865",
  "version": "501",
  "class": "patch",
  "metadata": {
    "title": "\nELSA-2024-20865: edk2 security update (MODERATE)\n",
    "affected": {
      "family": "unix",
      "platform": [
        "Oracle Linux 8"
      ]
    },
    "reference": [
      {
        "source": "elsa",
        "ref_id": "ELSA-2024-20865",
        "ref_url": "https://linux.oracle.com/errata/ELSA-2024-20865.html"
      },
      {
```

Reply:
Use title for the source 9259ed6